src/PIL/Image.py

```diff
@@ -480,16 +480,26 @@ class Image:
 
     def __init__(self):
         # FIXME: take "new" parameters / other image?
-        # FIXME: turn mode and size into delegating properties?
         self.im = None
-        self.mode = ""
+        self._mode = ""
         self._size = (0, 0)
         self.palette = None
         self.info = {}
         self.readonly = 0
         self.pyaccess = None
         self._exif = None
 
+    @property
+    def mode(self):
+        return self.im.mode if self.im else self._mode
+
+    @mode.setter
+    def mode(self, value):
+        if self.im:
+            self.im.mode = value
```

Comment:
It's intended that you can't directly set size (or mode for now), `setter`s are not needed.

Reply:
The problem is that you *can* currently set the mode and size, as can be seen by the many failing tests, so not allowing this would be a breaking change.

Reply:
There is no way to set size (only _size). The same rules should be for mode

Reply:
I created #7307 to prevent users from setting the mode directly.

Reply:
I've created https://github.com/Yay295/Pillow/pull/9 to fix the tests that fail here.